pkg/config/config.go

```diff
@@ -0,0 +1,320 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/iver-wharf/wharf-core/v2/pkg/config"
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Config holds all configurable settings for wharf-api.
+//
+// The config is read in the following order:
+//
+// 1. File: ~/.config/iver-wharf/wharf-cmd/wharf-cmd-config.yml
+//
+// 2. File: ./wharf-cmd-config.yml
+//
+// 3. File from environment variable: WHARF_CONFIG
+//
+// 4. Environment variables, prefixed with WHARF
+//
+// Each inner struct is represented as a deeper field in the different
+// configurations. For YAML they represent deeper nested maps. For environment
+// variables they are joined together by underscores.
+//
+// All environment variables must be uppercased, while YAML files are
+// case-insensitive. Keeping camelCasing in YAML config files is recommended
+// for consistency.
+type Config struct {
+	Worker         WorkerConfig
+	Provisioner    ProvisionerConfig
+	ProvisionerAPI ProvisionerAPIConfig
+	Aggregator     AggregatorConfig
+}
+
+func (c Config) print() {
+	data, err := yaml.Marshal(&c)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%s\n", string(data))
+}
+
+// WorkerConfig holds settings for the worker.
+type WorkerConfig struct {
+	K8s   K8sConfig
+	Steps StepsConfig
+}
+
+// K8sConfig holds settings for when the provisioner is using kubernetes.
+type K8sConfig struct {
+	// Context is the context used when talking to kubernetes.
+	//
+	// Added in v0.8.0.
+	Context string
+
+	// Namespace is the kubernetes namespace to talk to.
+	//
+	// Added in v0.8.0.
+	Namespace string
+}
+
+// StepsConfig holds settings for the different types of steps.
+type StepsConfig struct {
+	Docker  DockerStepConfig
+	Kubectl KubectlStepConfig
+	Helm    HelmStepConfig
+}
+
+// DockerStepConfig holds settings for the docker step type.
+type DockerStepConfig struct {
+	// Image is the image for the kaniko executor to use in the docker step.
+	//
+	// Added in v0.8.0.
+	Image string
+
+	// ImageTag is the image tag to use for the image.
+	//
+	// Added in v0.8.0.
+	ImageTag string
+}
+
+// KubectlStepConfig holds settings for the kubectl step type.
+type KubectlStepConfig struct {
+	// Image is the image to use in the kubectl step.
+	//
+	// Added in v0.8.0.
+	Image string
+
+	// ImageTag is the image tag to use for the image.
+	//
+	// Added in v0.8.0.
+	ImageTag string
+}
+
+// HelmStepConfig holds settings for the helm step type.
+type HelmStepConfig struct {
+	// Image is the image to use in the helm step.
+	//
+	// Added in v0.8.0.
```

Comment:
nit: Would be good with a note of why we don't have an image tag here, such as:

```suggestion
	// Image is the image to use in the helm step.
	//
	// There's no config value for the Docker image tag to use, as
	// that's defined by the helmVersion field in the helm step type
	// from the .wharf-ci.yml file.
	//
	// Added in v0.8.0.
```